src/serializer/ResidualHeapSerializer.js

```diff
@@ -1855,14 +1851,10 @@ export class ResidualHeapSerializer {
           let delayReason;
           if (instantRenderMode) {
             if (this.emitter.getReasonToWaitForDependencies(propValue)) {
-              let error = new CompilerDiagnostic(
-                "InstantRender does not yet support cyclical arays or objects",
-                val.expressionLocation,
-                "PP0039",
-                "FatalError"
+              this.realm.instantRenderBailout(
+                "InstantRender does not yet support cyclical arrays or objects",
```

Comment:
Why is this no longer a fatal error? Is it likely that the code generated in this case will do the right thing? If so, I want to see a regression test that does the right thing in this case.